x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/LookupJoin.java

```diff
@@ -71,4 +75,14 @@ protected NodeInfo<Join> info() {
             config().rightFields()
         );
     }
+
+    @Override
+    public void postAnalysisVerification(Failures failures) {
+        super.postAnalysisVerification(failures);
+        if (right() instanceof EsRelation esr) {
```

Comment:
Rather than directly checking `right`, can we traverse down and look for the `EsRelation`?

I'm afraid we might stuff something else between the `EsRelation` and the lookup join some time in the future, and then this check will be accidentally circumvented.